service/github/check.go

```diff
@@ -0,0 +1,342 @@
+	loc := c.Diagnostic.GetLocation()
+	startLine := int(loc.GetRange().GetStart().GetLine())
+	endLine := int(loc.GetRange().GetEnd().GetLine())
+	if endLine == 0 {
+		endLine = startLine
+	}
+	a := &github.CheckRunAnnotation{
+		Path:            github.String(loc.GetPath()),
+		StartLine:       github.Int(startLine),
+		EndLine:         github.Int(endLine),
+		AnnotationLevel: github.String(ch.annotationLevel(c.Diagnostic.Severity)),
+		Message:         github.String(c.Diagnostic.GetMessage()),
+		Title:           github.String(ch.buildTitle(c)),
+	}
+	// Annotations only support start_column and end_column on the same line.
+	if startLine == endLine {
+		if s, e := loc.GetRange().GetStart().GetColumn(), loc.GetRange().GetEnd().GetColumn(); s != 0 && e != 0 {
+			a.StartColumn = github.Int(int(s))
+			a.EndColumn = github.Int(int(e))
+		}
+	}
+	if s := c.Diagnostic.GetOriginalOutput(); s != "" {
+		a.RawDetails = github.String(s)
+	}
+	return a
+}
+
+func (ch *Check) buildTitle(c *filter.FilteredDiagnostic) string {
+	var sb strings.Builder
+	toolName := c.Diagnostic.GetSource().GetName()
+	if toolName == "" {
+		toolName = ch.ToolName
+	}
+	if toolName != "" {
+		sb.WriteString(fmt.Sprintf("[%s] ", toolName))
+	}
+	loc := c.Diagnostic.GetLocation()
+	sb.WriteString(loc.GetPath())
+	if startLine := int(loc.GetRange().GetStart().GetLine()); startLine > 0 {
+		sb.WriteString(fmt.Sprintf("#L%d", startLine))
+		if endLine := int(loc.GetRange().GetEnd().GetLine()); startLine < endLine {
+			sb.WriteString(fmt.Sprintf("-L%d", endLine))
+		}
+	}
+	if code := c.Diagnostic.GetCode().GetValue(); code != "" {
+		if url := c.Diagnostic.GetCode().GetUrl(); url != "" {
+			sb.WriteString(fmt.Sprintf(" <%s>(%s)", code, url))
+		} else {
+			sb.WriteString(fmt.Sprintf(" <%s>", code))
+		}
+	}
+	return sb.String()
+}
+
+func (ch *Check) postAnnotations(ctx context.Context, checkID int64, annotations []*github.CheckRunAnnotation) error {
+	opt := github.UpdateCheckRunOptions{
+		Name: ch.checkName(),
+		Output: &github.CheckRunOutput{
+			Title:       github.String(ch.checkTitle()),
+			Summary:     github.String(""), // Post summary with the last request.
+			Annotations: annotations[:min(maxAnnotationsPerRequest, len(annotations))],
+		},
+	}
+	if _, _, err := ch.CLI.Checks.UpdateCheckRun(ctx, ch.Owner, ch.Repo, checkID, opt); err != nil {
+		return err
+	}
+	if len(annotations) > maxAnnotationsPerRequest {
+		return ch.postAnnotations(ctx, checkID, annotations[maxAnnotationsPerRequest:])
+	}
+	return nil
+}
+
+// https://developer.github.com/v3/checks/runs/#parameters-1
+func (ch *Check) conclusion(annotations []*github.CheckRunAnnotation) string {
+	checkResult := "success"
+
+	if ch.Level != "" {
+		// Level takes precedence when configured (for backwards compatibility)
+		if len(annotations) == 0 {
+			return checkResult
+		}
+		switch strings.ToLower(ch.Level) {
+		case "info", "warning":
+			return "neutral"
+		}
+		return "failure"
+	} else {
```

Comment:
**[golint-circleci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>if block ends with a return statement, so drop this else and outdent its block
<!-- __reviewdog__:ChA1ODM1MzIxNzBkMTA1NDgwEg9nb2xpbnQtY2lyY2xlY2k= -->